monkey/infection_monkey/exploit/wmiexec.py

```diff
@@ -21,98 +24,79 @@ class WmiExploiter(HostExploiter):
-            lm_hash_hashed = self._config.hash_sensitive_data(lm_hash)
-            ntlm_hash_hashed = self._config.hash_sensitive_data(ntlm_hash)
-            creds_for_logging = (
-                "user, password (SHA-512), lm hash (SHA-512), ntlm hash (SHA-512): "
-                "({},{},{},{})".format(user, password_hashed, lm_hash_hashed, ntlm_hash_hashed)
-            )
-            logger.debug(
-                ("Attempting to connect %r using WMI with " % self.host) + creds_for_logging
-            )
+            creds_for_log = _get_credential_string([user, password, lm_hash, ntlm_hash])
+            logger.debug(f"Attempting to connect to {self.host} using WMI with {creds_for_log}")
 
             wmi_connection = WmiTools.WmiConnection()
 
             try:
                 wmi_connection.connect(self.host, user, password, None, lm_hash, ntlm_hash)
             except AccessDeniedException:
                 self.report_login_attempt(False, user, password, lm_hash, ntlm_hash)
-                logger.debug(
-                    ("Failed connecting to %r using WMI with " % self.host) + creds_for_logging
-                )
+                logger.debug(f"Failed connecting to {self.host} using WMI")
                 continue
             except DCERPCException:
                 self.report_login_attempt(False, user, password, lm_hash, ntlm_hash)
-                logger.debug(
-                    ("Failed connecting to %r using WMI with " % self.host) + creds_for_logging
-                )
+                logger.debug(f"Failed connecting to {self.host} using WMI")
                 continue
+
             except socket.error:
-                logger.debug(
-                    ("Network error in WMI connection to %r with " % self.host) + creds_for_logging
-                )
-                return False
+                logger.debug(f"Network error in WMI connection to {self.host}")
+                return self.exploit_result
+
             except Exception as exc:
                 logger.debug(
-                    ("Unknown WMI connection error to %r with " % self.host)
-                    + creds_for_logging
-                    + (" (%s):\n%s" % (exc, traceback.format_exc()))
+                    f"Unknown WMI connection error to {self.host}: "
+                    f"{exc} {traceback.format_exc()}"
                 )
-                return False
+                return self.exploit_result
 
             self.report_login_attempt(True, user, password, lm_hash, ntlm_hash)
+            self.exploit_result.exploitation_success = True
 
             # query process list and check if monkey already running on victim
             process_list = WmiTools.list_object(
                 wmi_connection,
                 "Win32_Process",
                 fields=("Caption",),
-                where="Name='%s'" % ntpath.split(src_path)[-1],
+                where="Name='{0}'".format(self.options["dropper_target_path_win_64"]),
```

Comment:
```suggestion
                where=f"Name='{self.options["dropper_target_path_win_64"]}'",
```

Reply:
Doesn't work or at least confuses the IDE, because of the nested quotes

Reply:
It was fixed in https://github.com/guardicore/monkey/pull/1766/commits/7de128cd9b9ec37866c3209c551f68d1a8ec85aa